### PR TITLE
Allow hostname (e.g., k8s pod name) to be used as the cookie value.

### PR DESCRIPTION
**Pull request by brianloss**
For my use case, it is useful to have the kubernetes pod name come through in the cookie value (it makes some debugging tasks easier). This pull request adds an annotation that allows the behavior of the dynamic cookie to be turned off if DNS server resolving isn't in use.
I also fixed some problems with log messages that the go 1.11 compiler warned me about.

### CODE REVIEW

**Comment by brianloss** [comment]
Rebased to master, but would like this to go in the 0.7 release if that's possible.

File: README.md [hunk 1]
```diff
@@ -122,6 +123,10 @@ Configure if HAProxy should maintain client requests to the same backend server.
 * `ingress.kubernetes.io/affinity`: the only supported option is `cookie`. If declared, clients will receive a cookie with a hash of the server it should be fidelized to.
 * `ingress.kubernetes.io/session-cookie-name`: the name of the cookie. `INGRESSCOOKIE` is the default value if not declared.
 * `ingress.kubernetes.io/session-cookie-strategy`: the cookie strategy to use (insert, rewrite, prefix). `insert` is the default value if not declared.
+* `ingress.kubernetes.io/session-cookie-dynamic`: indicates whether or not dynamic cookie naming will be used. With the
+default of `true`, a cookie will be generated based on the ip address and port of the server, along with any specified
```
Comment (jcmoraisjr) [inline README.md]: ... a cookie will be generated as a hash of the ip address and port of ... (something like this)

File: rootfs/etc/haproxy/template/haproxy.tmpl [hunk 2]
```diff
@@ -140,9 +140,11 @@ backend {{ $backend.Name }}
 {{- end }}
 {{- $sticky := $backend.SessionAffinity }}
 {{- if eq $sticky.AffinityType "cookie" }}
-    cookie {{ $sticky.CookieSessionAffinity.Name }} {{ $sticky.CookieSessionAffinity.Strategy }} {{ if eq $sticky.CookieSessionAffinity.Strategy "insert" }}indirect nocache httponly{{ end }} dynamic
+    cookie {{ $sticky.CookieSessionAffinity.Name }} {{ $sticky.CookieSessionAffinity.Strategy }} {{ if eq $sticky.CookieSessionAffinity.Strategy "insert" }}indirect nocache httponly{{ end }}{{ if $sticky.CookieSessionAffinity.Dynamic }} dynamic{{ end }}
+{{ if $sticky.CookieSessionAffinity.Dynamic }}
```
Comment (jcmoraisjr) [inline rootfs/etc/haproxy/template/haproxy.tmpl]: Missing a `{{-` here.

File: rootfs/etc/haproxy/template/haproxy.tmpl [hunk 3]
```diff
@@ -165,6 +167,9 @@ backend {{ $backend.Name }}
             {{- if ge $slot.BackendEndpoint.Weight 0 }} weight {{ $slot.BackendEndpoint.Weight }}{{ end }}
         {{- else }} 127.0.0.1:81 disabled{{ end }}
         {{- template "backend" map $cfg $backend }}
+        {{- if eq $sticky.AffinityType "cookie" }}
+            {{- if not $sticky.CookieSessionAffinity.Dynamic }} cookie {{ $slot.BackendServerName }}{{- end }}
```
Comment (jcmoraisjr) [inline rootfs/etc/haproxy/template/haproxy.tmpl]: This last `{{ end }}` might be without `-`.

**Comment by brianloss** [comment]
> Hi, thanks again for supporting this controller. Following a few suggestions.
> 
> This PR is somewhat big and v0.7 is being stabilized. I cannot merge these changes, sorry. Otoh v0.8 will have a very little impact with new features if using with v0.7 controller - v0.8 has most of the controller rewritten from scratch and v0.7 implementation and updates will be available. About a month from now I hope to start v0.8 snapshots which will have a very stable v0.7 controller with your changes and almost nothing more. Hope this helps.

That's a bummer, but if it's at least included in the next release, I can stop maintaining a fork eventually.

File: README.md (hunk 1 above)
Comment (brianloss) [reply README.md]: Fixed.

File: rootfs/etc/haproxy/template/haproxy.tmpl (hunk 2 above)
Comment (brianloss) [reply rootfs/etc/haproxy/template/haproxy.tmpl]: Fixed.

File: rootfs/etc/haproxy/template/haproxy.tmpl (hunk 3 above)
Comment (brianloss) [reply rootfs/etc/haproxy/template/haproxy.tmpl]: Fixed.

**Comment by jcmoraisjr** [comment]
Merging, thanks!

> That's a bummer, but if it's at least included in the next release, I can stop maintaining a fork eventually.

I'm doing my best on behalf of stability. I need to draw the line somewhere otherwise we wouldn't have a stable release. What I'm planning to do from now is to release often, this should also help to see new code on stable images earlier.

**Pull request merged**